File: JavaScript/165-Compare Version Numbers.js
/**
 * @param {string} version1
 * @param {string} version2
 * @return {number}
 */
var compareVersion = function(version1, version2) {
    let arr1 = version1.split("."), arr2 = version2.split(".");
    
    while (arr1.length && arr2.length){
        let int1 = Number(arr1.shift()), int2 = Number(arr2.shift());
        if (int1 > int2){
            return 1;
        }else if(int2 > int1){
            return -1
        }
    }
    while (arr1.length){
        let int1 = Number(arr1.shift());
        if (int1 > 0) return 1;
    }
    
    while (arr2.length){
        let int2 = Number(arr2.shift());
        if (int2 > 0) return -1;
    }
    
    return 0;
};

File: C++/165-Compare Version Numbers.cpp
class Solution {
public:
    vector<string> split(const string& str, char delimiter) {
        vector<string> tokens;
        istringstream iss(str);
        string token;
        while (getline(iss, token, delimiter)) {
            tokens.push_back(token);
        }
        return tokens;
    }
    int compareVersion(string version1, string version2) {
        vector<string> arr1 = split(version1, '.');
        vector<string> arr2 = split(version2, '.');
        int ind1 = 0, ind2 = 0;

        while (ind1 < arr1.size() && ind2 < arr2.size()) {
            int i1 = stoi(arr1[ind1]), i2 = stoi(arr2[ind2]);
            if (i1 < i2) {
                return -1;
            } else if (i1 > i2) {
                return 1;
            }
            ind1++;ind2++;
        }

        while (ind1 < arr1.size()) {
            int i1 = stoi(arr1[ind1]);
            if (i1 > 0) return 1;
            ind1++;
        }

        while (ind2 < arr2.size()) {
            int i2 = stoi(arr2[ind2]);
            if (i2 > 0) return -1;
            ind2++;
        }

        return 0;
    }
};